flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/utils/FlowNodeUtil.java

```diff
@@ -55,4 +55,17 @@ public static boolean withinDoClause(NodeBuilder.TemplateContext context) {
         return CommonUtils.withinDoClause(context.workspaceManager(), context.filePath(),
                 context.codedata().lineRange());
     }
+
+    /**
+     * Get the property key for a given key. If the key is a reserved property key, append an underscore to it.
+     *
+     * @param key the key to check
+     * @return the property key
+     */
+    public static String getPropertyKey(String key) {
+        if (Property.RESERVED_PROPERTY_KEYS.contains(key)) {
```

Comment:
Won't this conflict the parameters of a function like below

```ballerina
function(string connection, string connection_) {
}
```

Instead, can we include an invalid identifier like `$` at the first? 
connection -> $connection
\$connection -> \$connection